From 81ec3f0aa228345b31f4845d789f65fad4631f00 Mon Sep 17 00:00:00 2001
From: Teo Hall <teo.hall@nxp.com>
Date: Tue, 24 Jul 2018 15:18:35 -0500
Subject: [PATCH 4302/5242] MLK-19034 clk: imx8qm: Fix clk_unused crash

commit  ba122f4f586faf8d83b9805413dc6bf7e81fd3d2 from
https://source.codeaurora.org/external/imx/linux-imx.git

Remove unused ROMCP clks and related as LPCG
no longer exists

Signed-off-by: Teo Hall <teo.hall@nxp.com>
Reviewed-by: Anson Huang <Anson.Huang@nxp.com>
(cherry picked from commit 1c15332dffe7e41f0b9d367b96dd426798ec8b06)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/imx/clk-imx8qm.c   |    3 ---
 include/soc/imx8/imx8qm/lpcg.h |    1 -
 2 files changed, 4 deletions(-)

diff --git a/drivers/clk/imx/clk-imx8qm.c b/drivers/clk/imx/clk-imx8qm.c
index 6afa41e..a3248de 100644
--- a/drivers/clk/imx/clk-imx8qm.c
+++ b/drivers/clk/imx/clk-imx8qm.c
@@ -559,9 +559,6 @@ static int imx8qm_clk_probe(struct platform_device *pdev)
 	clks[IMX8QM_FSPI1_IPG_S_CLK] = imx_clk_gate2_scu("fspi1_ipg_s_clk", "lsio_bus_clk_root", LPCG_ADDR(FSPI_1_LPCG), 0x18, FUNCTION_NAME(PD_LSIO_FSPI_1));
 	clks[IMX8QM_FSPI1_IPG_CLK] = imx_clk_gate2_scu("fspi1_ipg_clk", "fspi1_ipg_s_clk", LPCG_ADDR(FSPI_1_LPCG), 0x14, FUNCTION_NAME(PD_LSIO_FSPI_1));
 	clks[IMX8QM_FSPI1_CLK] = imx_clk_gate_scu("fspi_1_clk", "fspi_1_div", SC_R_FSPI_1, SC_PM_CLK_PER, LPCG_ADDR(FSPI_1_LPCG), 0, 0);
-	clks[IMX8QM_ROMCP_REG_CLK] = imx_clk_gate2_scu("romcp_reg_clk", "lsio_bus_clk_root", LPCG_ADDR(ROMCP_LPCG), 0x10, FUNCTION_NAME(PD_LSIO));
-	clks[IMX8QM_ROMCP_CLK] = imx_clk_gate2_scu("romcp_clk", "lsio_mem_clk_root", LPCG_ADDR(ROMCP_LPCG), 0x0, FUNCTION_NAME(PD_LSIO));
-	clks[IMX8QM_96KROM_CLK] = imx_clk_gate2_scu("96krom_clk", "lsio_mem_clk_root", LPCG_ADDR(ROMCP_LPCG), 0x4, FUNCTION_NAME(PD_LSIO));
 
 	clks[IMX8QM_LSIO_MU5A_IPG_S_CLK] = imx_clk_gate2_scu("mu5_a_ipg_s_clk", "lsio_bus_clk_root", LPCG_ADDR(MU_5A_LPCG), 0x10, FUNCTION_NAME(PD_LSIO_MU5A));
 	clks[IMX8QM_LSIO_MU5A_IPG_CLK] = imx_clk_gate2_scu("mu5_a_ipg_clk", "mu5_a_ipg_s_clk",  LPCG_ADDR(MU_5A_LPCG), 0x0, FUNCTION_NAME(PD_LSIO_MU5A));
diff --git a/include/soc/imx8/imx8qm/lpcg.h b/include/soc/imx8/imx8qm/lpcg.h
index c41d54d..a22e7a2 100644
--- a/include/soc/imx8/imx8qm/lpcg.h
+++ b/include/soc/imx8/imx8qm/lpcg.h
@@ -25,7 +25,6 @@
 #define		GPIO_5_LPCG		0x5D4D0000
 #define		GPIO_6_LPCG		0x5D4E0000
 #define		GPIO_7_LPCG		0x5D4F0000
-#define		ROMCP_LPCG		0x5D500000
 #define		FSPI_0_LPCG		0x5D520000
 #define		FSPI_1_LPCG		0x5D530000
 #define		GPT_0_LPCG		0x5D540000
-- 
1.7.9.5

